### PR TITLE
feat: add functions_firebase_auth

### PR DESCRIPTION
**Pull request by grant**
Adds default Firebase Auth GCF handler.

### CODE REVIEW

File: functions/snippets/src/main/java/FirebaseAuth.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+// [START functions_firebase_auth]
+import com.google.cloud.functions.Context;
+import com.google.cloud.functions.RawBackgroundFunction;
+import java.util.logging.Logger;
+
+public class FirebaseAuth implements RawBackgroundFunction {
+
+  private static final Logger logger = Logger.getLogger(FirebaseAuth.class.getName());
```
Comment (kurtisvg) [inline functions/snippets/src/main/java/FirebaseAuth.java]:
```suggestion
  private static final Logger LOGGER = Logger.getLogger(FirebaseAuth.class);
```

File: functions/snippets/src/main/java/FirebaseAuth.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Override
+  public void accept(String json, Context context) {
```
Comment (kurtisvg) [inline functions/snippets/src/main/java/FirebaseAuth.java]: nit: Perhaps you should show the user how to parse the JSON?

File: functions/snippets/src/main/java/FirebaseAuth.java (hunk 1 above)
Comment (grant) [reply functions/snippets/src/main/java/FirebaseAuth.java]: Done.

File: functions/snippets/src/main/java/FirebaseAuth.java (hunk 2 above)
Comment (grant) [reply functions/snippets/src/main/java/FirebaseAuth.java]: That would change the functionality of the snippet and would make Java different than the other languages.
I've added a stub for parsing JSON though.

**Pull request merged**